### PR TITLE
add dataset class for mimic-cxr jpg dataset

### PR DESCRIPTION
**Pull request by jennyzhang-petuum**

### CODE REVIEW

**Comment by christy-yuan-li** [comment]
The data augmentation / transformation is currently set as None. Could you add some standard image augmentation? Besides, what’s the difference between the 'transform' and 'data_aug' argument in the MIMICCXR_Dataset init function? 

Change line106 to relative paths. For example, add the data folder "/media/files/datasets/physionet.org/files/mimic-cxr-jpg/2.0.0/“ as an args parameter. 

Could you try to change the print function (Line 132) to a visualization function that plots the read tensors back to images? That way we could check the data loader pipeline very easily. 

Line84: are the dataset images at grey scale (instead of RGB)? 

Line89: for this case, do we skip this image or continue to read?

File: data/MIMIC_CXR.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                            "Lung Lesion",
+                            "Edema",
+                            "Consolidation",
+                            "Pneumonia",
+                            "Atelectasis",
+                            "Pneumothorax",
+                            "Pleural Effusion",
+                            "Pleural Other",
+                            "Fracture",
+                            "Support Devices"]
```
Comment (hunterhector) [inline data/MIMIC_CXR.py]: We can change this to a `__init__` parameter too, maybe.

Comment (hunterhector) [reply data/MIMIC_CXR.py]: In fact, I recommend using Texar's `hyperparam` system: https://github.com/asyml/texar-pytorch/blob/master/texar/torch/data/data/paired_text_data.py#L226

File: data/MIMIC_CXR.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    valid_dataset.csv = dataset.csv.iloc[valid_dataset.indices]
+    train_loader = torch.utils.data.DataLoader(train_dataset,
+                                               batch_size=1,
+                                               shuffle=True,
+                                               num_workers=0,
+                                               pin_memory=True,
+                                               drop_last=True)
+
+    for batch in train_loader:
+        print(batch)
```
Comment (hunterhector) [inline data/MIMIC_CXR.py]: Newline at the end is a good practice

**Comment by jennyzhang-petuum** [comment]
Yep. I agree. I will add texar hyperparam

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: data/MIMIC_CXR.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+
+class MIMICCXR_Dataset:
+    """
+
+    Dataset website here:
+    https://physionet.org/content/mimic-cxr-jpg/2.0.0/
+    """
+
+    def __init__(self, imgpath, csvpath, metacsvpath, views=["PA"], transform=None, data_aug=None,
```
Comment (hunterhector) [inline data/MIMIC_CXR.py]: mutable default argument here will create memory leaks and other problems:

https://docs.python-guide.org/writing/gotchas/#what-you-wrote

**Comment by jennyzhang-petuum** [comment]
The dataset images are at grey scale (instead of RGB).

File: report-generation-nlp.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
+import dgl
```
Comment (hunterhector) [inline report-generation-nlp.py:1]: Actually I think you should open another PR for this, so that we solve a smaller problem one at a time

**Pull request closed without merging**